products/iam/api.yaml

```diff
@@ -20,6 +20,9 @@ versions:
   - !ruby/object:Api::Product::Version
     name: ga
     base_url: https://iam.googleapis.com/v1/
+  - !ruby/object:Api::Product::Version
```

Comment:
Interesting- Magic Modules assumes that subsequent APIs are a rough superset of each other, but this `v1beta` API doesn't contain the other handful of IAM resources at all. We may need to represent this as a separate entry in the `products/` folder, and override it in `terraform.yaml` to name it correctly.

You don't need that yet- I want to see what changes the generator makes, exactly.

Reply:
Ah, yep,  we'll want to add this as a new `products/` entry. MM assumes that each version in an API is a rough superset of more stable versions (ga < beta < alpha < eap) because it uses a single API endpoint for all resources under a product. IAM exposes some beta-only resources at a beta endpoint but doesn't also have the GA resources there.

You'll want to create a new `products/` folder- probably `iambeta`- and create an api.yaml and terraform.yaml there. The `name` (line 21 in this file) should also likely be `IAMBeta`. Inside `terraform.yaml`, you can override the default resource name using https://github.com/GoogleCloudPlatform/magic-modules/blob/master/overrides/terraform/resource_override.rb#L34-L38 so that the resources are `google_iam_*` not `google_iam_beta_*`

This should result in a second IAM basepath/endpoint getting registered, which points to the beta endpoint.